compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -515,7 +515,7 @@ fn make_code_region(
         // Extend an empty span by one character so the region will be counted.
         let CharPos(char_pos) = start_col;
         if span.hi() == body_span.hi() {
-            start_col = CharPos(char_pos - 1);
+            start_col = CharPos(char_pos.saturating_sub(1));
```

Comment:
I'm wondering if we should use saturating math in the other operations in this function since it's not much code? Specifically, can we also fix the addition on lines 520, 531 and 533?

Reply:
@tmandry - Do you mind doing that, if you agree it's worth it?

Sorry for the bug, but I thought `lookup_file_pos()` should never return less than 1. Maybe I misunderstood or the implementation changed, but that's not something the compiler would have enforced of course, so this is a good fix. Thanks.